doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,226 @@
+## Multiple y cable vendor API design
+
+### Overview
+
+This document summarizes the approach taken to accommodate multiple
+Y cable vendors for supporting dual-ToR configurations in SONiC.
+All the design choices are presented
+such that vendors can implement the Y cable interface with ease and effeciency
+and all the feature requirements are met.
+
+
+Challenge: to provide an interface for Y cable to interact with PMON docker and HOST
+           which can be uniform across all vendors.
+
+### Directory Structure
+
+#### Background
+
+- We need to have an appropriate place where vendors can implement their modules/packages and provide Sonic with an abstraction layer where pmon and host can access the Y cable packages of all the vendors with ease
+   
+  - Currently the assumption is Y cable package implementation would be in python itself
```

Comment:
Change to `Y-cable API will be written in the Python language`

Reply:
fixed